File: android/app/src/main/java/com/coinswallet/coins/TonCoin.java
// EthereumCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;

import wallet.core.jni.CoinType;
import wallet.core.jni.PrivateKey;

public class TonCoin extends CoinFactory.Coin {
    private final String address;
    private final PrivateKey privateKey;

    public TonCoin(String mnemonic) {
        super(mnemonic);
        this.privateKey = wallet.getKeyForCoin(CoinType.TON);
        this.address = CoinType.TON.deriveAddress(this.privateKey);
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        return this.address;
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        return CoinFactory.Coin.toHex(this.privateKey.data());
    }
    public String signTransaction(String rawData) {
        return null;
    }
}

File: ios/TonCoin.swift
//
//  EthereumCoin.swift
//  coinswallet
//
//  Created by Tal Grynbaum on 30/06/2023.
//

import Foundation
import WalletCore

class TonCoin: CoinFactory.Coin {
  private var address: String = ""
  private var privateKey: String = ""
  
  override init(mnemonic: String) {
    super.init(mnemonic: mnemonic)
    let privateKeyInfo = wallet.getKeyForCoin(coin: .ton)
    self.address = CoinType.ton.deriveAddress(privateKey: privateKeyInfo)
    self.privateKey = privateKeyInfo.data.hexString;
  }
  
  override func getNewAddress(isTestNet:Bool) -> String {
    return self.address
  }
  
  override func getPrivateKey(isTestNet:Bool) -> String {
    return self.privateKey;
  }
  
  override func signTransaction(rawData: String) -> String {
    // Implement the logic to sign an Ethereum transaction using the provided raw data
    return ""
  }
}
